pkgs/native_assets_cli/lib/src/model/build_output.dart

```diff
@@ -76,13 +84,12 @@ final class BuildOutputImpl implements BuildOutput {
         'Flutter, please update native_assets_cli.',
       );
     }
-
-    final assets =
-        AssetImpl.listFromJsonList(as<List<Object?>>(jsonMap[_assetsKey]));
-
     return BuildOutputImpl(
       timestamp: DateTime.parse(as<String>(jsonMap[_timestampKey])),
-      assets: assets,
+      assets: AssetImpl.listFromJson(as<List<Object?>>(jsonMap[_assetsKey])),
+      assetsForLinking: as<Map<String, dynamic>>(jsonMap[_assetsForLinkingKey])
```

Comment:
Shouldn't this be `as<Map<String, dynamic>?>` to be backwards compatible?

Older `build.dart`s will not output this key.

Please add a test to exercise this.

Rev the protocol version to 1.3.0 and document this.